src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,102 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [state, setState] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const handleInputChange = (name: string, value: string) => {
+    setState((prevState) => ({
+      ...prevState,
+      [name]: value,
+    }));
+  };
+
+  const resetFormValues = () => {
+    setState({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const newMovie: Movie = {
+      title: state.title,
+      description: state.description,
+      imgUrl: state.imgUrl,
+      imdbUrl: state.imdbUrl,
+      imdbId: state.imdbId,
+    };
+
+    onAdd(newMovie);
+
+    setCount((prevCount) => prevCount + 1);
+    resetFormValues();
+  };
+
+  const isAddButtonEnabled
+  = state.title.trim()
+    && state.imdbUrl.trim()
+    && state.imgUrl.trim()
+    && state.imdbId.trim();
 
   return (
-    <form className="NewMovie" key={count}>
+    <form className="NewMovie" key={count} onSubmit={handleSubmit}>
```

Comment:
```suggestion
    <form
      className="NewMovie"
      key={count}
      onSubmit={handleSubmit}
    >
```
Don't forget about code style